File: code_samples/sample.java
package com.adeso;

import java.io.IOException;

import com.adeso.domain.Theme;

class Vampire {
  private String location;
  public int birthDate = 1902;
  private int deathDate;
  private int[] deathDates;
  private String[] weaknesses = "Hello";
  private boolean alive = true;
  private Boolean hungry = false;

  public Vampire(String location, int birthDate, int deathDate, String[] weaknesses) {
    this.location = location;
    this.birthDate = birthDate;
    this.deathDate = deathDate;
    this.weaknesses = weaknesses;
  }

  public int getAge() {
    return this.calcAge();
  }

  public int calcAge() {
    return this.deathDate - this.birthDate;
  }
}

// ...there was a guy named Vlad
public class dracula {
  dracula() {
    super();
  }

  private boolean mybool = true;
  Vampire vampire = new Vampire(
      "Transylvania",
      1428,
      1476,
      new String[] { "Sunlight", "Garlic" });

  private Date date = vampire.birthDate;

  private doSomething(){
      if (13>=5){
        vampire.calcAge();        
      }

  // Código en Java
  for(String key:invalidControls.keySet())
  {
    this.errorMessages.get(key).put("message", this.labels.errors.VA0006);
    this.errorMessages.get(key).put("hasError", true);
    if(exception != null && exception.errors.length > 0){
      System.out.println("hello");
    }

  }
 }
}

File: code_samples/sample.py
import re

class MyClass:
    def __init__(self):
        self.my_variable = 1
        self.my_constant = 2

    def my_method(self):
        print("Hello World")

    def my_arrow_function(self, x):
        return x * 2

    def my_interface(self):
        pass

class MyOtherClass(MyClass):
    def __init__(self):
        super().__init__()

    def my_method(self):
        print("Hello Universe")

my_object = MyOtherClass()
print(my_object.my_variable)
print(my_object.my_constant)
my_object.my_method()
print(my_object.my_arrow_function(3))


class FileReader:
    def __init__(self, filename):
        self.filename = filename

    def read(self):
        with open(self.filename) as f:
            return f.read()

class WordCounter:
    def __init__(self, text):
        self.text = text

    def count_words(self):
        pattern = r'\b[A-Z]\w*\b'
        words = re.findall(pattern, self.text)
        counts = {}
        for word in words:
            if word in counts:
                counts[word] += 1
            else:
                counts[word] = 1
        return counts

if __name__ == '__main__':
    reader = FileReader('example.txt')
    text = reader.read()
    counter = WordCounter(text)
    counts = counter.count_words()
    for word, count in counts.items():
        print(f"{word}: {count}")





class Vampire:
  def __init__(self, props):
    self.location = props['location']
    self.birthDate = props['birthDate']
    self.deathDate = props['deathDate']
    self.weaknesses = props['weaknesses']
    

  def get_age(self):
    return self.calc_age()

  def calc_age(self):
    return self.deathDate - self.birthDate
  
  for key in invalidControls.keys():
    this.errorMessages[key]["message"] = this.labels.errors.VA0006
    this.errorMessages[key]["hasError"] = True
    if exception is not None and len(exception.errors) > 0:
        print("hello")



# ...there was a guy named Vlad

Dracula = Vampire({
  'location': 'Transylvania',
  'birthDate': 1428,
  'deathDate': 1476,
  'weaknesses': ['Sunlight', 'Garlic']
})

lightSensible = True